test/test.js

```diff
@@ -184,6 +184,26 @@ describe('Page', function() {
       await neverResolves.catch(e => error = e);
       expect(error.message).toContain('Protocol error');
     }));
+    it('should fire page closed event', SX(async function() {
+      const newPage = await browser.newPage();
+      let fired = false;
+      newPage.on('closed', closed => {
+        fired = true;
+        expect(closed).toBe('page closed');
+      });
+      await newPage.close();
+      expect(fired).toBe(true);
+    }));
+  });
+
+  describe('Page.Events.closed', function(){
+    it('should emmited when page got crashed', SX(async function() {
+      let emitted = false;
+      page.on('closed', () => emitted = true);
+      page.goto('chrome://crash');
```

Comment:
you should be able to await `page.goto` here, instead of doing the `waitForEvents` on the following line

Reply:
it fail on my machine if i do `await page.goto(url);`

Reply:
[here](https://travis-ci.org/GoogleChrome/puppeteer/jobs/284151708#L525) the travis test failure. That's why i wait on error, maybe you can assist me?